controllers/applicationvolumereplication_controller.go

```diff
@@ -195,658 +276,895 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
 		return ctrl.Result{}, errorswrapper.Wrap(err, "failed to get AVR object")
 	}
 
-	subscriptionList := &subv1.SubscriptionList{}
-	listOptions := []client.ListOption{
-		client.InNamespace(avr.Namespace),
-		client.MatchingLabels(avr.Spec.SubscriptionSelector.MatchLabels),
-	}
-
-	err = r.Client.List(ctx, subscriptionList, listOptions...)
+	avrPlRule, userPlRule, err := r.getPlacementRules(ctx, avr)
 	if err != nil {
-		if !errors.IsNotFound(err) {
-			logger.Error(err, "failed to find subscription list", "namespace", avr.Namespace)
+		logger.Error(err, "failed to get PlacementRules")
 
-			return ctrl.Result{Requeue: true}, nil
-		}
+		return ctrl.Result{}, err
+	}
+
+	// Make sure that we give time to the cloned PlacementRule to run and produces decisions
+	if len(avrPlRule.Status.Decisions) == 0 {
+		const initialWaitTime = 5
 
-		return ctrl.Result{}, errorswrapper.Wrap(err, "failed to list subscriptions")
+		return ctrl.Result{RequeueAfter: time.Second * initialWaitTime}, nil
```

Comment:
(future) Can we not own the plRule and hence get triggered automatically once it is reconciled?

Also, why do we need a decision unless we are in the dynamic initial placement (i.e no preferredCluster), should we check that and then wait at an appropriate time (say during initialPlacement)?